src/Lecture3.hs

```diff
@@ -125,10 +131,10 @@ data Reward = Reward
     } deriving (Show, Eq)
 
 instance Semigroup Reward where
-
+    Reward x s1 <> Reward y s2 = Reward (Gold (unGold x + unGold y)) (s1 || s2)
 
 instance Monoid Reward where
-
+    mempty = Reward (Gold 0) False
```

Comment:
You can use `mempty` for `Gold` here because it already has a `Monoid` instance 🏅 
```suggestion
    mempty = Reward mempty False
```